specification/batch/Batch/routes.tsp

```diff
@@ -0,0 +1,2755 @@
+  @header
+  `request-id`?: string;
+}
+
+@doc("The parameters for a widget status request")
+model BatchApplicationListHeaders extends BatchClientRequestHeaders {
+  @doc("""
+The maximum number of items to return in the response. A maximum of 1000
+applications can be returned.
+""")
+  @query
+  @minValue(1)
+  @maxValue(1000)
+  @extension(
+    "x-ms-parameter-grouping",
+    {
+      postfix: "Options",
+    }
+  )
+  maxresults?: int32 = 1000;
+
+  #suppress "@azure-tools/typespec-azure-core/casing-style" "The names of Model types must use camelCase"
+  @doc("""
+The time the request was issued. Client libraries typically set this to the
+current system clock time; set it explicitly if you are calling the REST API
+directly.
+""")
+  @header
+  @format("date-time-rfc1123")
+  @extension(
+    "x-ms-parameter-grouping",
+    {
+      postfix: "Options",
+    }
+  )
+  `ocp-date`?: string;
+}
+
+@doc("Common header parms for Pool related File operartions")
+model BatchPoolFileClientRequestHeaders extends BatchClientRequestHeaders {
+  @doc("The ID of the Pool that contains the Compute Node.")
+  @path
+  poolId: string;
+
+  @doc("The ID of the Compute Node from which you want to delete the file.")
+  @path
+  nodeId: string;
+
+  @doc("The path to the file or directory that you want to delete.")
+  @path
+  filePath: string;
+}
+
+@doc("Common header parms for Job related File operartions")
+model BatchJobFileClientRequestHeaders extends BatchClientRequestHeaders {
+  @doc("The ID of the Job that contains the Task.")
+  @path
+  jobId: string;
+
+  @doc("The ID of the Task whose file you want to retrieve.")
+  @path
+  taskId: string;
+
+  @doc("The path to the Task file that you want to get the content of.")
+  @path
+  filePath: string;
+}
+
+#suppress "@azure-tools/typespec-azure-core/casing-style" "The names of Model types must use camelCase"
+@doc("The parameters for a widget status request")
+model BatchClientRequestHeaders {
+  @doc("""
+The maximum number of items to return in the response. A maximum of 1000
+applications can be returned.
+""")
+  @query
+  @extension(
+    "x-ms-parameter-grouping",
+    {
+      postfix: "Options",
+    }
+  )
+  timeOut?: int32 = 30;
+
+  @doc("""
+The caller-generated request identity, in the form of a GUID with no decoration
+such as curly braces, e.g. 9C4D50EE-2D56-4CD3-8152-34347DC9F2B0.
+""")
+  @header
+  @extension(
+    "x-ms-parameter-grouping",
+    {
+      postfix: "Options",
+    }
+  )
+  @format("uuid")
+  `client-request-id`?: string;
+
+  @doc("Whether the server should return the client-request-id in the response.")
+  @header
+  @extension(
+    "x-ms-parameter-grouping",
+    {
+      postfix: "Options",
+    }
+  )
+  `return-client-request-id`?: boolean = false;
```

Comment:
My concern: is it possible for service to default add `client-request-id` if it is set in request?

Reply:
Why do we need this header?